pkg/collector/labels.go

```diff
@@ -28,9 +30,11 @@ func Labels(instance v1alpha1.OpenTelemetryCollector) map[string]string {
 			base[k] = v
 		}
 	}
+	version := strings.Split(instance.Spec.Image, ":")
 
 	base["app.kubernetes.io/managed-by"] = "opentelemetry-operator"
 	base["app.kubernetes.io/instance"] = naming.Truncate("%s.%s", 63, instance.Namespace, instance.Name)
+	base["app.kubernetes.io/version"] = version[len(version)-1]
```

Comment:
Could this cause a panic if the length of version is empty? as well as in other places in the patch.

Reply:
I'm not sure that `version` can be empty here, I think it would include at least an empty string, but it can certainly be length 1 which would not produce the expected result.  The length of the list should definitely be checked and `latest` used as a default if it is not 2.

Reply:
Thank you @Aneurysm9 and @frzifus for your consideration. I'll check it.